File: 2.PrefexSumAndBinarySearch/python/binary_search.py
import sys 

input_data = sys.stdin.read().strip().splitlines()

n,q = map(int,input_data[0].split(" "))
a = sorted(map(int,input_data[1].split(" ")))

for i in range(0,q):
    x = int(input_data[2+i])
    l = 0
    r = n-1
    flag = False
    while l<=r:
        middle = (l+r)//2
        if a[middle] == x:
            flag=True
            break
        elif x > a[middle]:
            l=middle+1
        elif x < a[middle]:
            r=middle-1
    if flag==True:
        print('found')
    else :
        print('not found')

File: 2.PrefexSumAndBinarySearch/typescript/root/binary_search.ts
import * as readline from 'readline'

const rl = readline.createInterface({
    input:process.stdin
})

const input_data:string[] = []

rl.on('line',(line)=>{
    input_data.push(line)
})
rl.on('close',()=>{
    const [n,q]:number[] = input_data[0].split(" ").map(Number)
    const a = input_data[1].split(" ").map(Number).sort((a:number,b:number)=>a-b)
    
    for (let i:number=0;i<q;i++){
        const x = parseInt(input_data[2+i])
        let l = 0;
        let r = n-1;
        let flag = false
        while (l<=r){
            const middle = Math.floor((l+r)/2)
            if (a[middle]===x){
                flag=true;
                break
            }else if (x > a[middle]){
                l=middle+1;
            }else if (x < a[middle]){
                r=middle-1;
            }
        }
        if (flag===true){
            console.log("found")
        } else {
            console.log("not found")
        }
    }
})